### PR TITLE
Updated `ruff pre-commit` version and modified few lines according to it.

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
#### Ruff 0.0.285 has been released (https://github.com/astral-sh/ruff-pre-commit).
So, I have updated the ruff version in `.pre-commit-config.yaml` file, ran `ruff .` and corrected few files according to the errors it showed.

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by Sai-Suraj-27** [comment]
@mroeschke sir, can you please review the PR, Thank you.

File: pandas/core/internals/construction.py [hunk 1]
```diff
@@ -903,7 +903,7 @@ def _list_of_dict_to_arrays(
 
     # assure that they are of the base dict class and not of derived
     # classes
-    data = [d if type(d) is dict else dict(d) for d in data]
+    data = [d if type(d) is dict else dict(d) for d in data]  # noqa: E721
```
Comment (twoertwein) [inline pandas/core/internals/construction.py:906]: Instead of ignoring the issue, maybe try `[d if isinstance(d, dict) ...]`?

Comment (Sai-Suraj-27) [reply pandas/core/internals/construction.py:906]: @twoertwein yes, sir. So, I guess there is nothing I need to change?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Sai-Suraj-27

**Pull request merged**